reciprocalspaceship/algorithms/intensity.py

```diff
@@ -0,0 +1,94 @@
+import numpy as np
+
+import reciprocalspaceship as rs
+
+
+def compute_intensity_from_structurefactor(
+    ds,
+    F_key,
+    SigF_key,
+    output_columns=None,
+    dropna=True,
+    inplace=False,
+):
+    """
+    Back-calculate intensities and approximate intensity error estimates from
+    structure factor amplitudes and error estimates
+
+    Intensity computed as I = SigF*SigF + F*F. Intensity error estimate
+    approximated as SigI = abs(2*F*SigF)
+
+    Parameters
+    ----------
+    ds : DataSet
+        Input DataSet containing columns with F_key and SigF_key labels
+    F_key : str
+        Column label for structure factor amplitudes to use in back-calculation
+    SigF_key : str
+        Column label for structure factor error estimates to use in
+        back-calculation
+    output_columns : list or tuple of column names
+        Column labels to be added to ds for recording back-calculated I and
+        SigI, respectively. output_columns must have len=2.
+    dropna : bool
+        Whether to drop reflections with NaNs in F_key or SigF_key
+        columns
```

Comment:
Since this error propagation only depends on individual Miller indices, it seems to me that including NaNs shouldn't be problematic for the calculation. If there are NaNs in the input, there will be corresponding NaNs in the output. Am I missing something here? I'm not sure whether it's useful to sanitize the input here if it doesn't harm the calculation.

Reply:
I will run this through `pytest` to confirm but I believe you're right here